src/Lighthouses.java

```diff
@@ -0,0 +1,170 @@
+
+import java.util.Arrays;
+import java.util.Scanner;
+
+/* og�lnie je�li user wybierze pole z latarni� to nie mo�na mu na to pozwoli�, czyli je�li wstawi statek
+ * w pole z latarni� musi wyst�pi� wyj�tek (je�li pole do wstawienia statku b�dzie r�ne od '_'')
+ * trzeba b�dzie parstowa� zapisane pola na int �eby sprawdzi� ile statk�w o�wietl�
+ * trzeba zabroni� userowi wpisania innego znaku ni� int*/
+public class Lighthouses {
+
+    char board[][] = new char[12][12];
+    Scanner key = new Scanner(System.in);
+
+    public static void main( String[] args ) {
+        Scanner key = new Scanner(System.in);
+        Lighthouses lighthouses = new Lighthouses();
+        lighthouses.setBoard();
+        System.out.println("Tak wygl�da tablica do kt�rej trzeba wstawi� statki:");
+        System.out.println("'_' - to puste pole na kt�rym mo�na ustawi� statek.");
+        System.out.println("Cyfra w polu oznacza ile s�siednich p�l w pionie i poziomie o�wietla latarnia");
+        lighthouses.printBoard(lighthouses.board);
+        lighthouses.putShips(lighthouses.board);
+    }
+
+    public void printBoard( char board[][] ) {
+        System.out.println("    1  2  3  4  5  6  7  8  9  10 ");
+        for (int i = 1; i < board.length - 1; i++) {
+            if (i < 10) {
+                System.out.print(i + " :");
+            } else {
+                System.out.print(i + ":");
+            }
+            for (int j = 1; j < board[i].length - 1; j++) {
+                System.out.print("[" + board[i][j] + "]");
+            }
+            System.out.println(" ");
+        }
+    }
+
+    public void setBoard() {
+        board[1][1] = '1';
+        board[1][10] = '2';
+        board[3][8] = '4';
+        board[4][6] = '2';
+        board[5][4] = '0';
+        board[6][7] = '2';
+        board[7][5] = '0';
+        board[8][3] = '0';
+        board[10][1] = '2';
+        board[10][10] = '3';
+        for (int i = 0; i < board.length; i++) {
+            for (int j = 0; j < board[i].length; j++) {
+                if (board[i][j] == 0) {
+                    board[i][j] = '_';
+                }
+            }
+        }
+    }
+
+    public void putShips( char[][] tab ) {
+        boolean check = false;
+        System.out.println("Podaj koordynaty kom�rki w kt�r� chcesz wstawi� statek:");
+        System.out.println("Wpisz 'q' aby zako�czy� gr�, 'z' aby zdj�� statek, 's' aby sprawdzi� plansz�\n'd'aby i�� dalej");
+        char quit = key.next().charAt(0);
+        if (quit == 'z') {
+            removeShip(board);
+        } else if (quit == 's') {
```

Comment:
gdy masz sekwencję sprawdzającą cały czas tą samą zmienną, a interesuje Cię wartość.... czyli np.

quit == 'a' , to coś
quit == 'b' , to coś
quit == 'c' , to coś

to idealnym rozwiązaniem będzie switch, przechodzi on do konkretnej implementacji, nie sprawdzając wszystkich warunków po kolei.

zamień to na SWITCH.